### PR TITLE
[INS-406] Braintrust detector

### PR DESCRIPTION
**Pull request by MuneebUllahKhan222**
### Description

This PR adds the **Braintrust API Key Detector** for TruffleHog.  
It scans for Braintrust API keys (prefix `sk-`) and optionally verifies them via the official API.

**Regex**: `\b(sk-[A-Za-z0-9]{48})\b`

### Verification
For verification, we use the Braintrust Projects API: https://api.braintrust.dev/v1/project?limit=1. We send a GET request with the token in the Authorization: Bearer <token> header. A response code of 200 OK means the token is valid. 401 Unauthorized means it is an invalid or revoked token, while 403 Forbidden indicates a valid token with insufficient permissions.

This API endpoint is part of the official Braintrust API and can be used safely for verification. It is read-only and does not perform any destructive actions.

### Corpora Test
The detector does not appear in the list.
image

### Checklist:
* [x] Tests passing (`make test-community`)?
* [x] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/welcome/install/#local-installation))?

---

> [!NOTE]
> **Low Risk**
> Additive detector-only change with read-only verification; gated like other new detectors and covered by tests.
> 
> **Overview**
> Adds a **Braintrust API key** detector so scans can find `sk-` + 48-character tokens and optionally confirm them against Braintrust’s API.
> 
> The new scanner matches `\b(sk-[A-Za-z0-9]{48})\b`, pre-filters on the `braintrust` keyword, and when verification is on calls `GET https://api.braintrust.dev/v1/project?limit=1` with a Bearer token (200/403 → verified, 401 → not verified). Unit and integration tests cover regex edge cases and live verification behavior.
> 
> Wiring follows other gated detectors: proto enum `BrainTrustApiKey` (1053), `BraintrustDetectorEnabled` defaulting to on in `main.go`, registration in `buildDetectorList()`, and `BrainTrustApiKey` listed under feature-flag exclusions in `defaults_test.go`.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit c6c51f8a8a0df6e2a853b93d07c728d661528487. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: pkg/detectors/braintrust/braintrust.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	// Format: sk- + 48 alphanumeric characters (observed)
+	braintrustTokenPat = regexp.MustCompile(
+		`\b(sk-[A-Za-z0-9]{48})\b`,
+	)
+)
+
+// Keywords used for fast pre-filtering
+func (s Scanner) Keywords() []string {
+	return []string{"sk-"}
+}
```
Comment (shahzadhaider1) [inline pkg/detectors/braintrust/braintrust.go:36]: A couple of things to flag regarding the use of `sk-` as a detector keyword:

1. Keyword length: We generally avoid using prefixes as keywords unless they're at least 4-5 characters long or distinctive enough to avoid false positives. Since `sk-` is only three characters, it's better to rely on more specific, contextually relevant keywords instead.
2. False positive risk: The `sk-` prefix is shared across many different secret types, which means using it as a keyword would likely generate a high volume of false positives.

What do you think?

Comment (MuneebUllahKhan222) [reply pkg/detectors/braintrust/braintrust.go:36]: Initially, I thought that only OpenAI admin keys used the `sk-` prefix. However, further investigation showed that multiple AI/LLM providers also use keys starting with `sk-`.  

To reduce false positives, the keyword has now been made more specific.

File: pkg/detectors/braintrust/braintrust.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	regexp "github.com/wasilibs/go-re2"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct {
+	client *http.Client
+	detectors.DefaultMultiPartCredentialProvider
```
Comment (shahzadhaider1) [inline pkg/detectors/braintrust/braintrust.go]: This detector has a single credential

File: pkg/detectors/braintrust/braintrust.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+	}
+}
+
+func (s Scanner) Type() detectorspb.DetectorType {
+	return detectorspb.DetectorType_BrainTrustApiKey
+}
+
+func (s Scanner) Description() string {
+	return "Braintrust is an AI evaluation and observability platform. This detector identifies Braintrust API keys."
+}
```
Comment (shahzadhaider1) [inline pkg/detectors/braintrust/braintrust.go:135]: The `Description()` text is surfaced to customers in the enterprise UI, so mentioning "This detector identifies..." can be a bit confusing in that context. The pattern we follow is:
1. First sentence: What the service/platform is.
2. Second sentence: What risk the exposed credential poses.

Something like: _"Braintrust is an AI evaluation and observability platform. Braintrust API keys can be used to access and manage projects, experiments, and evaluation data."_
(Feel free to adjust the second sentence based on what the key actually grants access to.)

**Review by shahzadhaider1 — APPROVED** [review]

**Review by camgunz — APPROVED** [review]

**Review by amanfcp — APPROVED** [review]
The wiring LGTM
